installation-bundle/src/Config/ParameterDumper.php

```diff
@@ -34,17 +34,24 @@ class ParameterDumper
 
     public function __construct(string $rootDir, Filesystem $filesystem = null)
     {
-        $this->rootDir = $rootDir;
         $this->filesystem = $filesystem ?: new Filesystem();
         $parameters = [];
 
-        foreach (['app/config/parameters.yml.dist', 'app/config/parameters.yml'] as $file) {
+        if (file_exists($rootDir.'/config/parameters.yml') || !file_exists($rootDir.'/app/config/parameters.yml')) {
```

Comment:
This should be `if (is_dir($rootDir.'/config'))`.

Reply:
The directory can exist, but it might not contain the configuration file. For example the Contao Manager might write it's configuration there (see `ManagerConfig`), but the app config is still in `app/config`.

Reply:
I don't think we should support this use case. If there is a `/config` directory, all config files should be in there.

Reply:
Then we would need to adapt this change to the `ManagerConfig` and the Contao Manager needs to if/else on the directories as well?

Reply:
Yes. Splitting config files across multiple directories will only confuse people.

Reply:
What if there is a config directory in the root for any other reason? The application would suddenly no longer boot… I really don't like this idea. We should either force everyone to the root config directory, or we should correctly support both cases. And a (potentially empty) directory does not tell us whether that directory contains the app configuration.

Reply:
That's right. So then here is my preferred logic:

```php
if (file_exists($rootDir.'/app/config/parameters.yml')) {
    @trigger_error('…');
    $this->configDir = $rootDir.'/app/config';
} else {
    $this->configDir = $rootDir.'/config';
}
```

I then expect all other configuration files, including the `contao-manager.yml` file, to be in the same folder as the `parameters.yml` file.

Reply:
This would choose the old folder if you copy everything to the root (instead of moving it).

Reply:
Correct.

Reply:
And that makes sense to you? Didn't we agree to always use the root folder if it exists (see the voting)?

Reply:
Yeah, but that is not what your code does! For instance, if there is a `config/parameters.yml` file and an `app/config/routing.yml` file, both will be loaded.

Reply:
No, that has been changed a while ago. Please read the PR changes again 🙃

Reply:
I have. `elseif (file_exists($configFile = $this->rootDir.'/app/config/routing.yml')) {` loads the file if there is no `routing.yml` file in the `config` folder.

I think it is best to retarget the PR to Contao 4.9 and discuss this thoroughly before implementing anything.

Reply:
How is the `routing.yml` now related to your statement ?

> Yeah, but that is not what your code does! For instance, if there is a config/parameters.yml file and an app/config/routing.yml file, both will be loaded.

Reply:
We have agreed on always using the root folder if it exists (see voting). But your code still loads the `routing.yml` file from `app/config` even if the root folder exists.

Reply:
After discussing this on Slack, we realized that the [voting](https://github.com/contao/contao/pull/566#issuecomment-509570073) was misleading and that we actually do want to support both folders; with the files in `config` taking precedence over the ones in `app/config`.